src/hooks/use-breakpoint.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+export const windowSizes = {
+  sm: 640,
+  md: 768,
+  lg: 1024,
+  xl: 1280,
+  '2xl': 1536,
+};
+
+export const useBreakpoint = (size: keyof typeof windowSizes) => {
+  const [isWindowLessSize, setIsWindowLessSize] = useState(false);
+  useEffect(() => {
+    function handleResize() {
+      if (window.innerWidth < windowSizes[size]) {
```

Comment:
I would suggest implementing a check using `>=` instead, to match `@media (min-width: *)` behaviour used in css classes